editoast/src/core/mod.rs

```diff
@@ -84,11 +84,17 @@ impl CoreClient {
         Self::Direct(client)
     }
 
-    pub async fn new_mq(uri: String, worker_pool_identifier: String, timeout: u64) -> Result<Self> {
+    pub async fn new_mq(
+        uri: String,
+        worker_pool_identifier: String,
+        timeout: u64,
+        single_worker: bool,
+    ) -> Result<Self> {
```

Comment:
Not convinced by the benefits: on the only caller place, we would just embed the `mq_client::Options` creation, revealing more of the internals (and adding little more code/import).
Not strictly opposed to it, but not so fond of the idea here.

Reply:
I'll be helpful to settle the debate: I'm fine with both :p

Reply:
Is revealing the internal really a problem? It will make calling the function more explicit. Compare these, for example:
```rust
new_mq("foobar", "yolo", 42, false);
```

As a reader of the code, I will therefore need to go consult the function to understand what each of these parameters mean (which in Github might not be trivial, so I may need to pull the code, to benefit from `rust-analyzer`).

Compared to this:
```rust
new_mq(mq_client::Options {
  uri: "foobar",
  worker_pool_identifier: "yolo",
  timeout: 42,
  single_worker: false
});
```

Sure it's more verbose, but more verbose and more explicit it's not necessarily bad if you think that code is read more often than it's written. In some languages, you can name the parameter when calling but sadly, Rust doesn't allow that ([there has been discussion about it](https://internals.rust-lang.org/t/pre-rfc-named-arguments/16413/1)).

⚠️ This comment is long, because I wanted to be explicit about why I tend to be averse to a lot of arguments in a function. For functions with only few arguments, usually, the name of the function is enough to understand what the 1 or 2 arguments are, without naming them. That said, 4 is not yet **a lot of arguments**, just enough to trigger me. So I'm fine to keep it as it is, feel free to resolve whatever you do.

Reply:
Convinced by the named params: https://github.com/OpenRailAssociation/osrd/pull/9166/commits/52ffad3e68fc252d8e036c9ec76ba718383749ad#diff-09ec5a58d69869bd41dba8612f5c3973f7bbe0147510aa5f2a1a44632ef30903R417-R422